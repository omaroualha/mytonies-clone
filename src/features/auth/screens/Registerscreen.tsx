import React, { FC } from "react";
import { ModalWrapper, Webview } from "@/components";

export const RegisterScreen: FC = () => {
  const url =
    "https://login.tonies.com/auth/realms/tonies/protocol/openid-connect/registrations?client_id=tonie-studio&kc_locale=en&redirect_uri=com.tonies.app%3A%2Foauthredirect&response_type=code&scope=openid+email+profile&webview=app&utm_source=app&cookieConsent=16";
  return (
    <ModalWrapper>
      <Webview url={url} title="toniesÂ® - Create your Tonies-Account" />
    </ModalWrapper>
  );
};
