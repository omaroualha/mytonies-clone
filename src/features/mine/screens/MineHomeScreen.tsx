import { images } from "@/assets";
import { Box, Headline } from "@/components";
import React, { FC } from "react";
import { ImageBackground, ScrollView, StyleSheet } from "react-native";
import { useSafeAreaInsets } from "react-native-safe-area-context";
import { useTheme } from "styled-components";
import { MineItem, MineItemProps } from "../components/MineItem";
import { MineStackScreenProps } from "../MineStack";

export type MineHomeScreenProps = MineStackScreenProps<"Home">;

export const MineHomeScreen: FC<MineHomeScreenProps> = ({ navigation }) => {
  const theme = useTheme();
  const insets = useSafeAreaInsets();

  const mines: MineItemProps[] = [
    {
      headline: "Toniebox",
      image: images.mineTonieBox,
      description:
        "Currently, there is no Toniebox set up with your account. Add one by tapping on the +.",
    },
    {
      headline: "Tonies",
      image: images.mineTonies,
      description:
        "Your Tonies will appear here. Simply place them on your Toniebox connected to the Wi-fi.",
    },
    {
      headline: "Content",
      image: images.mineContent,
      description:
        "All your audio files will be displayed here. Record your own stories assign them to your Creative-tonies.",
    },
  ];
  return (
    <Box flex={1} backgroundColor={theme.colors.primary.light}>
      <Box height={insets.top} style={styles.blurView} />
      {/* <Box widt={"80%"} height="100%" flex={1} zIndex={10}> */}
      {/* <SVGShape /> */}
      {/* </Box> */}
      <ScrollView>
        <ImageBackground
          source={images.mineShapeBG}
          style={{
            width: 170,
            // zIndex: 10,
            height: 130,

            position: "absolute",
            right: 0,
          }}
        ></ImageBackground>

        <Box flex={1} pt={insets.top} px="L">
          <Box
            flexDirection={"row"}
            flex={1}
            // backgroundColor="yellow"
            justifyContent={"space-between"}
          >
            <Headline variant="HS">Mine</Headline>
          </Box>
          {mines.map((mine, index) => (
            <MineItem key={index} {...mine} />
          ))}
        </Box>
      </ScrollView>
    </Box>
  );
};

const styles = StyleSheet.create({
  blurView: {
    width: "100%",
    position: "absolute",
    backgroundColor: "rgba(255,255,255,0.95)",
    zIndex: 1,
    top: 0,
    left: 0,
    right: 0,
  },
});
