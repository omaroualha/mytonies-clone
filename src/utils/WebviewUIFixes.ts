export const uiFixesApp = (): string => {
  const styles = `
    .main-navigation{
      display: none;
    }
    .browserupgrade .SiteTheme__Site-sc-1ycpvs-0 kKWbES .sc-ehmTmK fstLMM SiteTheme__StyledMainNavigation-sc-1ycpvs-1 cdoBQQ {
      display: none;
    }
    .b-account-page-breadcrumb-wrapper {
      display: none;
    }
    `;
  const minifiedStyles = styles.replace(/\s\s+/g, "");

  return `;(function() {
      if (document.body.dataset.injectedFromApp) return;
      document.head.insertAdjacentHTML("beforeend", "<style>${minifiedStyles}</style>");
      document.body.dataset.injectedFromApp = 'true';
    })();
    `;
};
