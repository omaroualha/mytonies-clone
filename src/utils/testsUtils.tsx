import * as React from "react";
import { BaseTheme } from "@/theme/BaseTheme";
import { NavigationContainer } from "@react-navigation/native";
import { render } from "@testing-library/react-native";
import { ThemeProvider } from "styled-components/native";

/** Render helper that renders `ui` within `NavigationContainer`. */
const renderWithNavigation = (screen: JSX.Element) => {
  return render(
    <ThemeProvider theme={BaseTheme}>
      <NavigationContainer>{screen}</NavigationContainer>
    </ThemeProvider>
  );
};

const renderWithTheme = (component: JSX.Element) => {
  return render(<ThemeProvider theme={BaseTheme}>{component}</ThemeProvider>);
};

export const testsUtils = { renderWithNavigation, renderWithTheme };
