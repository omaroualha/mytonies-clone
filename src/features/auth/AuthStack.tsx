import React, { FC } from "react";
import {
  createNativeStackNavigator,
  NativeStackScreenProps,
} from "@react-navigation/native-stack";
import { CompositeScreenProps } from "@react-navigation/native";
import { RootStackScreenProps } from "../../navigation/RootStack";
import { LoginScreen } from "./screens/LoginScreen";
import { IntroScreen } from "./screens/IntroScreen";
import { RegisterScreen } from "./screens/Registerscreen";

export type AuthStackParamList = {
  Intro: undefined;
  Login: undefined;
  Register: undefined;
};

export type AuthStackScreenProps<Screen extends keyof AuthStackParamList> =
  CompositeScreenProps<
    NativeStackScreenProps<AuthStackParamList, Screen>,
    RootStackScreenProps<"AuthStack">
  >;

const Stack = createNativeStackNavigator<AuthStackParamList>();

export const AuthStack: FC<AuthStackScreenProps<"Intro">> = ({
  navigation,
}) => {
  return (
    <Stack.Navigator
      initialRouteName="Intro"
      screenOptions={{ headerShown: false }}
    >
      <Stack.Screen name="Intro" component={IntroScreen} />
      <Stack.Group
        screenOptions={{
          presentation: "transparentModal",
          gestureEnabled: false,
        }}
      >
        <Stack.Screen name="Login" component={LoginScreen} />
        <Stack.Screen name="Register" component={RegisterScreen} />
      </Stack.Group>
    </Stack.Navigator>
  );
};
