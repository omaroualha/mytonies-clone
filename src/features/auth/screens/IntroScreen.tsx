import React, { FC } from "react";
import { Image, StyleSheet, TouchableOpacity } from "react-native";
import { useTheme } from "styled-components/native";
import { MaterialIcons } from "@expo/vector-icons";
import { Box, Headline, Paragraph, Label, PrimaryButton } from "@/components";
import { AuthStackScreenProps } from "../AuthStack";
import { images } from "@/assets";
import { BaseTheme } from "@/theme/BaseTheme";

export type IntroScreenProps = AuthStackScreenProps<"Intro">;

export const IntroScreen: FC<IntroScreenProps> = ({ navigation }) => {
  const theme = useTheme();

  const handleLoginPress = () => {
    navigation.navigate("Login");
  };

  const handleRegisterPress = () => {
    navigation.navigate("Register");
  };

  return (
    <Box flex={1} backgroundColor={theme.colors.primary.brand}>
      <Box flex={2} width="100%" alignItems="center" justifyContent={"center"}>
        <Image source={images.toniesLogo} style={styles.logo} />
      </Box>
      <Box
        flex={1}
        width={"100%"}
        p="M"
        justifyContent={"space-between"}
        backgroundColor={theme.colors.primary.light}
        borderTopLeftRadius={theme.radii.L * 2}
        borderTopRightRadius={theme.radii.L * 2}
      >
        <Box flex={0.1} />
        <Box>
          <Headline variant="H2">Unlock your Tonies</Headline>
          <Label variant="L1" color={theme.colors.neutral.grey60}>
            Set up your Tonibox and discover the whole variety of your tonies
            with your mytonies acount.
          </Label>
        </Box>
        <Box pb="M">
          <PrimaryButton
            label="Log in"
            onPress={handleLoginPress}
            iconType="NEXT"
          />
          <PrimaryButton
            label="Free registration"
            onPress={handleRegisterPress}
            iconType="NEXT"
          />
        </Box>
      </Box>
    </Box>
  );
};

const styles = StyleSheet.create({
  logo: {
    width: "55%",
    height: "55%",
    resizeMode: "contain",
  },
  button: {
    height: 35,
    width: "100%",
    flexDirection: "row",
    marginTop: BaseTheme.space.S,
    borderRadius: BaseTheme.radii.M,
    backgroundColor: BaseTheme.colors.neutral.grey10,
    paddingLeft: BaseTheme.space.M,
    paddingRight: BaseTheme.space.M,
    justifyContent: "space-between",
    alignItems: "center",
  },
});
