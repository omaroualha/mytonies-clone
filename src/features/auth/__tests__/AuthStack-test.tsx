import React from "react";
import { testsUtils } from "@/utils/testsUtils";
import { fireEvent, screen } from "@testing-library/react-native";
import { IntroScreen } from "../screens/IntroScreen";
import { AuthStack } from "../AuthStack";

describe("AUTH STACK", () => {
  const navigation = { navigate: jest.fn() };
  const spy = jest.spyOn(navigation, "navigate");

  it(`Should match snapshot`, () => {
    testsUtils.renderWithNavigation(<AuthStack navigation={navigation} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  describe("INTRO SCREEN", () => {
    it(`Should renders correctly`, () => {
      testsUtils.renderWithNavigation(<IntroScreen navigation={navigation} />);
      expect(screen.getByText("Unlock your Tonies")).toBeTruthy();
      expect(screen.getByText("Log in")).toBeTruthy();
      expect(screen.getByText("Free registration")).toBeTruthy();
      fireEvent.press(screen.getByText("Log in"));
      fireEvent.press(screen.getByText("Free registration"));
      expect(spy).toBeCalledTimes(2);
    });

    describe("LOGIN SCREEN", () => {
      it(`Should renders correctly`, () => {
        testsUtils.renderWithNavigation(<AuthStack navigation={navigation} />);
        fireEvent.press(screen.getByText("Log in"));
        expect(screen.getByText("tonies® - Login")).toBeTruthy();
        expect(screen.getByTestId("close-button")).toBeTruthy();
        fireEvent.press(screen.getByTestId("close-button"));
        expect(spy).toBeCalledTimes(2);
      });
    });

    describe("REGISTER SCREEN", () => {
      it(`Should renders correctly`, () => {
        testsUtils.renderWithNavigation(<AuthStack navigation={navigation} />);
        fireEvent.press(screen.getByText("Free registration"));
        expect(
          screen.getByText("tonies® - Create your Tonies-Account")
        ).toBeTruthy();
        expect(screen.getByTestId("close-button")).toBeTruthy();
        fireEvent.press(screen.getByTestId("close-button"));
        expect(spy).toBeCalledTimes(2);
      });
    });
  });
});
