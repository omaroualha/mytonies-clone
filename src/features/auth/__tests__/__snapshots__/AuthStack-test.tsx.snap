// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AUTH STACK Should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Intro"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            backgroundColor="#c12822"
            flex={1}
            style={
              Array [
                Object {
                  "backgroundColor": "#c12822",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <View
              alignItems="center"
              flex={2}
              justifyContent="center"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexBasis": 0,
                    "flexGrow": 2,
                    "flexShrink": 1,
                    "justifyContent": "center",
                    "width": "100%",
                  },
                ]
              }
              width="100%"
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": "55%",
                    "resizeMode": "contain",
                    "width": "55%",
                  }
                }
              />
            </View>
            <View
              backgroundColor="#ffffff"
              borderTopLeftRadius={32}
              borderTopRightRadius={32}
              flex={1}
              justifyContent="space-between"
              p="M"
              style={
                Array [
                  Object {
                    "backgroundColor": "#ffffff",
                    "borderTopLeftRadius": 32,
                    "borderTopRightRadius": 32,
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "space-between",
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 16,
                    "width": "100%",
                  },
                ]
              }
              width="100%"
            >
              <View
                flex={0.1}
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexGrow": 0.1,
                      "flexShrink": 1,
                    },
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#222222",
                        "fontFamily": "Museo-700",
                        "fontSize": 20,
                        "fontWeight": "500",
                        "letterSpacing": 0,
                        "lineHeight": 32,
                      },
                    ]
                  }
                  variant="H2"
                >
                  Unlock your Tonies
                </Text>
                <Text
                  color="#64676e"
                  style={
                    Array [
                      Object {
                        "color": "#64676e",
                        "fontFamily": "Museo-300",
                        "fontSize": 12,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 18,
                      },
                    ]
                  }
                  variant="L1"
                >
                  Set up your Tonibox and discover the whole variety of your tonies with your mytonies acount.
                </Text>
              </View>
              <View
                pb="M"
                style={
                  Array [
                    Object {
                      "paddingBottom": 16,
                    },
                  ]
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#e5e5e5",
                      "borderRadius": 8,
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "marginTop": 8,
                      "minHeight": 35,
                      "opacity": 1,
                      "padding": 8,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#222222",
                          "fontFamily": "Museo-300",
                          "fontSize": 15,
                          "letterSpacing": 0,
                          "lineHeight": 18,
                        },
                        Object {
                          "color": "#222222",
                        },
                      ]
                    }
                    variant="P2"
                  >
                    Log in
                  </Text>
                  <Text />
                </View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#e5e5e5",
                      "borderRadius": 8,
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "marginTop": 8,
                      "minHeight": 35,
                      "opacity": 1,
                      "padding": 8,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#222222",
                          "fontFamily": "Museo-300",
                          "fontSize": 15,
                          "letterSpacing": 0,
                          "lineHeight": 18,
                        },
                        Object {
                          "color": "#222222",
                        },
                      ]
                    }
                    variant="P2"
                  >
                    Free registration
                  </Text>
                  <Text />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
